defuse/src/contract/intents/simulate.rs

```diff
@@ -1,30 +1,28 @@
-use std::collections::HashMap;
-
-use defuse_core::DefuseError;
-use defuse_core::error::Result;
+use defuse_core::events::DefuseEvent;
 use defuse_core::{
     Deadline,
     accounts::AccountEvent,
     engine::Inspector,
     intents::{
         IntentEvent,
-        token_diff::{TokenDeltas, TokenDiff},
-        tokens::{FtWithdraw, MtWithdraw, NativeWithdraw, NftWithdraw, StorageDeposit, Transfer},
+        token_diff::TokenDeltas,
+        tokens::{FtWithdraw, MtWithdraw, NftWithdraw},
     },
-    tokens::{Amounts, TokenId},
 };
-use defuse_map_utils::cleanup::DefaultMap;
 use defuse_near_utils::UnwrapOrPanicError;
-use near_sdk::{AccountId, AccountIdRef, CryptoHash, require};
+use near_sdk::{AccountId, AccountIdRef, CryptoHash, serde_json};
+use std::collections::HashMap;
 
 pub struct SimulateInspector {
     pub intents_executed: Vec<IntentEvent<AccountEvent<'static, ()>>>,
     pub min_deadline: Deadline,
     pub balance_diff: HashMap<AccountId, TokenDeltas>,
+    #[allow(dead_code)] // FIXME: remove
     pub wnear_id: AccountId,
     pub ft_withdrawals: Option<Vec<FtWithdraw>>,
     pub nft_withdrawals: Option<Vec<NftWithdraw>>,
     pub mt_withdrawals: Option<Vec<MtWithdraw>>,
+    pub events_emitted: Vec<String>,
```

Comment:
Make it more convenient for callers to avoid parsing strings as JSON:
```suggestion
    pub events: Vec<serde_json::Value>,
```

Reply:
Or it would be even better if we don't bind to JSON specifically, but use strongly-typed `Event` struct/enum...

Reply:
Push event objects presents a challenge because the values are mostly references. In fact, I changed some Cow::Borrow to Owned to make this work. If we get rid of references, then this is easily doable.